docs-content/getting-started/fullstack-frameworks/next-js/8_configuration.md

```diff
@@ -0,0 +1,140 @@
+```
+
+We recommend shipping source maps to your production server. Your client-side JavaScript is always public, and code decompilation tools are so powerful that obscuring your source code may not be helpful.
+
+Shipping source maps to production with Next.js is as easy as setting `productionBrowserSourceMaps: true` in your `nextConfig`. Alternatively, you can upload source maps directly to Highlight using our `withHighlightConfig` function.
+
+
+```javascript
+// next.config.js
+const { withHighlightConfig } = require('@highlight-run/next/server')
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+	productionBrowserSourceMaps: false
+}
+
+module.exports = withHighlightConfig(nextConfig)
+```
+
+You must export your `HIGHLIGHT_SOURCEMAP_UPLOAD_API_KEY` to your build process. If you're building and deploying with Vercel, try our [Highlight Vercel Integration](https://vercel.com/integrations/highlight) to inject `HIGHLIGHT_SOURCEMAP_UPLOAD_API_KEY` automatically.
+
+<AutoplayVideo description="Sourcemap upload api key" src="https://user-images.githubusercontent.com/878947/235971066-def65bcb-1580-4e8a-9d69-3e5556d51b27.webm" />
+
+## Instrumentation.ts
+
+Next.js comes out of the box instrumented for Open Telemetry, but support can be spotty depending on your runtime. This example Highlight implementation uses Next's [experimental instrumentation feature](https://nextjs.org/docs/advanced-features/instrumentation) to configure Open Telemetry on a Next.js server.
+
+1.  Turn on `experimental.instrumentationHook`. We've also turned on `productionBrowserSourceMaps` because Highlight is much easier to use with source maps.
+
+```javascript
+// next.config.js
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+	experimental: {
+		instrumentationHook: true,
+	},
+	productionBrowserSourceMaps: true,
+}
+
+module.exports = nextConfig
+```
+
+2. Create `instrumentation.ts` at the root of your project as explained in the [instrumentation guide](https://nextjs.org/docs/advanced-features/instrumentation). Call `registerHighlight` from within the exported `register` function:
+
+```javascript
+// instrumentation.ts
+import CONSTANTS from '@/app/constants'
+
+export async function register() {
+	// importing @highlight-run/next/server in the Edge runtime throws errors.
+	if (process.env.NEXT_RUNTIME === 'nodejs') {
+		const { registerHighlight } = await import('@highlight-run/next/server')
+
+		registerHighlight({
+			projectID: CONSTANTS.NEXT_PUBLIC_HIGHLIGHT_PROJECT_ID,
+			otlpEndpoint: CONSTANTS.NEXT_PUBLIC_HIGHLIGHT_OTLP_ENDPOINT,
+			serviceName: 'my-nextjs-backend',
+		})
+	}
+}
+
+```
+
+4. If you're using App Router, re-export `/instrumentation.ts` from `src/instrumentation.ts` to enable for App Router.
+
+```typescript
+// https://github.com/vercel/next.js/discussions/48273
+// src/instrumentation.ts:
+export { register } from '../instrumentation'
+```
```

Comment:
does this still apply, or should this be removed?

Reply:
I've gone back and forth on this one. I can kill it.